files/unarchive.py

```diff
@@ -89,46 +107,358 @@
 
 import re
 import os
+import stat
+import pwd
+import grp
+import datetime
+import time
+import binascii
 from zipfile import ZipFile
 
 # String from tar that shows the tar contents are different from the
 # filesystem
-DIFFERENCE_RE = re.compile(r': (.*) differs$')
+OWNER_DIFF_RE = re.compile(r': Uid differs$')
+GROUP_DIFF_RE = re.compile(r': Gid differs$')
+MODE_DIFF_RE = re.compile(r': Mode differs$')
+MODE_DIFF_RE = re.compile(r' is newer or same age.$')
```

Comment:
To be honest, these changes do not really make a difference (pun intended). I can remove those changes from this PR. This was an attempt to implement check-mode using gtar, but I think a native re-implementation is in order. Shall I remove those ?